sources/controllers/node_controller.rb

```diff
@@ -11,28 +9,35 @@ class NodeController < Sinatra::Base
   end
 
   # html view, sends stuff to /node as POST
-  get '/new' do
-    @node = Node.new
-    haml :node
-  end
-
-  post '/' do
-    Node.new(params[:node])
-  end
+  get '/new' do             #
+    @node = Node.new        # dafuq is this
+    haml :node              #
```

Comment:
generate a new Node object and pass that into the view where you can enter the new values. The form sends the values via POST to /node.